packages/universal-router-sdk/src/entities/protocols/pancakeswap.ts

```diff
@@ -0,0 +1,403 @@
+      } else {
+        addMixedSwap(planner, trade, this.options, payerIsUser, routerMustCustody, performAggregatedSlippageCheck)
+      }
+    }
+    // const ZERO_OUT: CurrencyAmount<Currency> = CurrencyAmount.fromRawAmount(sampleTrade.outputAmount.currency, 0)
+    // const minAmountOut: CurrencyAmount<Currency> = trades.reduce(
+    //   (sum, trade) => sum.add(minimumAmountOut(trade, options.slippageTolerance)),
+    //   ZERO_OUT,
+    // )
+
+    let minAmountOut = SmartRouter.minimumAmountOut(trades[0], this.options.slippageTolerance, trades[0].outputAmount)
+    // The router custodies for 3 reasons: to unwrap, to take a fee, and/or to do a slippage check
+    if (routerMustCustody) {
+      // If there is a fee, that percentage is sent to the fee recipient
+      // In the case where ETH is the output currency, the fee is taken in WETH (for gas reasons)
+      if (this.options.fee) {
+        const feeBips = BigInt(encodeFeeBips(this.options.fee.fee))
+        planner.addCommand(CommandType.PAY_PORTION, [
+          sampleTrade.outputAmount.currency.wrapped.address,
+          this.options.fee.recipient,
+          feeBips,
+        ])
+
+        // If the trade is exact output, and a fee was taken, we must adjust the amount out to be the amount after the fee
+        // Otherwise we continue as expected with the trade's normal expected output
+        if (this.type === TradeType.EXACT_OUTPUT) {
+          minAmountOut = minAmountOut.subtract(minAmountOut.multiply(feeBips).divide(10000))
+        }
+      }
+      // The remaining tokens that need to be sent to the user after the fee is taken will be caught
+      // by this if-else clause.
+      if (outputIsNative) {
+        planner.addCommand(CommandType.UNWRAP_WETH, [this.options.recipient, BigInt(minAmountOut.quotient.toString())])
+      } else {
+        planner.addCommand(CommandType.SWEEP, [
+          sampleTrade.outputAmount.currency.wrapped.address,
+          this.options.recipient,
+          BigInt(minAmountOut.quotient.toString()),
+        ])
+      }
+    }
+    if (inputIsNative && (this.type === TradeType.EXACT_OUTPUT || SwapRouter.riskOfPartialFill(trades))) {
+      // for exactOutput swaps that take native currency as input
+      // we need to send back the change to the user
+      planner.addCommand(CommandType.UNWRAP_WETH, [this.options.recipient, 0n])
+    }
+  }
+}
+
+// encode a v2 swap
+function addV2Swap(
+  planner: RoutePlanner,
+  trade: SmartRouterTrade<TradeType>,
+  options: PancakeSwapOptions,
+  routerMustCustody: boolean,
+  payerIsUser: boolean,
+  performAggregatedSlippageCheck: boolean,
+): void {
+  const amountIn = BigInt(SmartRouter.maximumAmountIn(trade, options.slippageTolerance).quotient.toString())
+  const amountOut = BigInt(SmartRouter.minimumAmountOut(trade, options.slippageTolerance).quotient.toString())
+
+  // V2 trade should have only one route
+  const route = trade.routes[0]
```

Comment:
When we add single command, only the first route is considered. Should be the same for `addV3Swap` as well